editor/components/page-attributes/parent.js

```diff
@@ -43,31 +37,30 @@ export function PageAttributesParent( { parent, postType, items, onUpdateParent
 	);
 }
 
-const applyConnect = connect(
-	( state ) => {
-		return {
-			postId: getCurrentPostId( state ),
-			parent: getEditedPostAttribute( state, 'parent' ),
-			postTypeSlug: getCurrentPostType( state ),
-		};
-	},
-	{
-		onUpdateParent( parent ) {
-			return editPost( { parent: parent || 0 } );
-		},
-	}
-);
+const applyWithSelect = withSelect( ( select ) => {
+	const { getPostType } = select( 'core' );
+	const { getCurrentPostId, getEditedPostAttribute } = select( 'core/editor' );
+	const postTypeSlug = getEditedPostAttribute( 'type' );
+	return {
+		postId: getCurrentPostId(),
+		parent: getEditedPostAttribute( 'parent' ),
+		postType: getPostType( postTypeSlug ),
+		postTypeSlug,
+	};
+} );
 
-const applyWithAPIDataPostType = withAPIData( ( props ) => {
-	const { postTypeSlug } = props;
+const applyWithDispatch = withDispatch( ( dispatch ) => {
+	const { editPost } = dispatch( 'core/editor' );
 	return {
-		postType: `/wp/v2/types/${ postTypeSlug }?context=edit`,
+		onUpdateParent( parent ) {
+			editPost( { parent: parent || 0 } );
```

Comment:
`return` is missing. Not sure it is ever used but noting the difference with the original code.

Reply:
Right, i think it's useless. We needed the return previously because it was not auto-bound to dispatch.